test/lib/users/ssh_authorized_keyring_test.rb

```diff
@@ -90,122 +90,179 @@ def authorized_keys_from_home(path)
 
-    before { FileUtils.mkdir_p(home) }
+    before do
+      FileUtils.mkdir_p(ssh_dir)
+
+      allow(Yast::Users::SSHAuthorizedKeysFile).to receive(:new).and_return(file)
+    end
+
     after { FileUtils.rm_rf(tmpdir) if File.exist?(tmpdir) }
 
-    context "if no keys are registered for the given home" do
+    context "when there are not registered keys for the given home" do
       it "does not try to write the keys" do
         expect(file).to_not receive(:save)
+
         keyring.write_keys
       end
     end
 
-    context "if some keys are registered for the given home" do
+    context "when there are registered keys for the given home" do
       let(:uid) { 1001 }
       let(:gid) { 101 }
       let(:home_dir_exists) { true }
       let(:ssh_dir_exists) { false }
 
       before do
-        allow(Yast::SCR).to receive(:Execute).and_call_original
         allow(Yast::SCR).to receive(:Read).and_call_original
+        allow(Yast::SCR).to receive(:Execute).and_call_original
+        allow(Yast::SCR).to receive(:Execute)
+          .with(Yast::Path.new(".target.remove"), authorized_keys_path)
+
         allow(Yast::FileUtils).to receive(:Exists).and_call_original
         allow(Yast::FileUtils).to receive(:Exists).with(ssh_dir).and_return(ssh_dir_exists)
         allow(Yast::FileUtils).to receive(:Exists).with(home).and_return(home_dir_exists)
-        allow(Yast::FileUtils).to receive(:IsDirectory).with(ssh_dir)
-          .and_return(true)
-        keyring.add_keys([key])
+        allow(Yast::FileUtils).to receive(:IsDirectory).with(ssh_dir).and_return(true)
+        allow(Yast::FileUtils).to receive(:Chmod).and_call_original
+
+        # Load some authorized_keys
+        FileUtils.cp_r(
+          FIXTURES_PATH.join("home", "user1", ".ssh", "authorized_keys"),
+          authorized_keys_path
+        )
+        keyring.read_keys
       end
 
-      it "writes the keys" do
-        keyring.write_keys
-        expect(File).to exist(authorized_keys_path)
+      context "but no new keys are added" do
+        it "does not write keys again" do
+          expect(Yast::SCR).to_not receive(:Execute)
+            .with(Yast::Path.new(".target.remove"), authorized_keys_path)
+          expect(file).to_not receive(:save)
+
+          keyring.write_keys
+        end
       end
 
-      it "SSH directory and authorized_keys inherits owner/group from home" do
-        allow(Yast::FileUtils).to receive(:GetOwnerUserID).with(home).and_return(uid)
-        allow(Yast::FileUtils).to receive(:GetOwnerGroupID).with(home).and_return(gid)
-        expect(Yast::FileUtils).to receive(:Chown).with("#{uid}:#{gid}", ssh_dir, false)
-        expect(Yast::FileUtils).to receive(:Chown)
-          .with("#{uid}:#{gid}", authorized_keys_path, false)
+      context "but new keys are added" do
+        before do
+          keyring.add_keys([key])
+        end
 
-        keyring.write_keys
-      end
+        it "deletes old authorized_keys file" do
```

Comment:
np: I know it is something from the past, but whether the file is removed or not (before recreating it) is basically an implementation detail.